spring-integration-cloudevents/src/main/java/org/springframework/integration/cloudevents/transformer/ToCloudEventTransformer.java

```diff
@@ -0,0 +1,415 @@
+		}
+
+		if (eventFormat != null) {
+			return MessageBuilder.withPayload(eventFormat.serialize(cloudEvent))
+					.copyHeaders(headers)
+					.setHeader(MessageHeaders.CONTENT_TYPE, eventFormat.serializedContentType())
+					.build();
+		}
+		HashMap<String, Object> messageMap = new HashMap<>(headers);
+		messageMap.put(MessageHeaders.CONTENT_TYPE, "application/cloudevents");
+		return CloudEventUtils.toReader(cloudEvent)
+				.read(new MessageBuilderMessageWriter(this.cloudEventPrefix, new MessageHeaders(messageMap)));
+	}
+
+	@Override
+	public String getComponentType() {
+		return "ce:to-cloudevent-transformer";
+	}
+
+	/**
+	 * Extract CloudEvent extensions from message headers based on pattern matching.
+	 * Iterate through all message headers and apply the configured extension patterns
+	 * to determine which headers should be included as CloudEvent extensions.
+	 * @param headers the message headers to extract extensions from
+	 * @return a map of header key-value pairs that match the extension patterns;
+	 *         returns an empty map if no headers match the patterns
+	 */
+	private Map<String, Object> getCloudEventExtensions(MessageHeaders headers) {
+		Map<String, Object> cloudEventExtensions = new HashMap<>();
+		Boolean patternResult;
+		for (Map.Entry<String, Object> header : headers.entrySet()) {
+			patternResult = PatternMatchUtils.smartMatch(header.getKey(), this.extensionPatterns);
+			if (patternResult != null && patternResult) {
+				cloudEventExtensions.put(header.getKey(), header.getValue());
+			}
+		}
+		return cloudEventExtensions;
+	}
+
+	private static class ToCloudEventTransformerExtension implements CloudEventExtension {
+
+		private final Map<String, Object> cloudEventExtensions;
+
+		/**
+		 * Create CloudEvent extensions by processing a message using expressions.
+		 * @param headers to be added as cloudEventExtensions
+		 */
+		ToCloudEventTransformerExtension(Map<String, Object> headers) {
+			this.cloudEventExtensions = Map.copyOf(headers);
+		}
+
+		@Override
+		public void readFrom(CloudEventExtensions extensions) {
+			throw new UnsupportedOperationException();
+		}
+
+		@Override
+		public @Nullable Object getValue(String key) throws IllegalArgumentException {
+			return this.cloudEventExtensions.get(key);
+		}
+
+		@Override
+		public Set<String> getKeys() {
+			return this.cloudEventExtensions.keySet();
+		}
+
+	}
+
+	private static class MessageBuilderMessageWriter implements CloudEventWriter<Message<byte[]>>,
+			CloudEventWriterFactory<MessageBuilderMessageWriter, Message<byte[]>> {
+
+		private final String cloudEventPrefix;
+
+		private final String specVersionKey;
+
+		private final Map<String, Object> headers = new HashMap<>();
+
+		/**
+		 * Create a MessageBuilderMessageWriter with the specified configuration.
+		 * @param cloudEventPrefix the prefix to prepend to CloudEvent attribute names in message headers
+		 * @param headers the base message headers to include in the output message
+		 */
```

Comment:
Same here: Javadoc on the `MessageBuilderMessageWriter` itself, not on the private ctor.